src/utils.jl

```diff
@@ -60,3 +63,45 @@ end
 # We want the AnyPlatform to look like `default_host_platform`,
 get_concrete_platform(::AnyPlatform, shards::Vector{CompilerShard}) =
     get_concrete_platform(default_host_platform, shards)
+
+function march_comparison_strategy(a::String, b::String, a_requested::Bool, b_requested::Bool)
+    # If both b and a requested, then we fall back to equality:
+    if a_requested && b_requested
+        return a == b
+    end
+
+    function get_arch_isa(isa_name::String)
+        for (arch, isas) in arch_march_isa_mapping
+            for (name, isa) in isas
+                name == isa_name && return arch, isa
+            end
+        end
+        return nothing, nothing
+    end
+
+    a_arch, a_isa = get_arch_isa(a)
+    b_arch, b_isa = get_arch_isa(b)
+    if any(isnothing, (a_arch, b_arch)) || a_arch != b_arch
+        # Architectures are definitely not compatible, exit early
+        return false
+    end
+
+    if a_requested
+        # ISA `b` is compatible with ISA `a` only if it's a subset of `a`
+        return b_isa ≤ a_isa
+    else
+        # ISA `a` is compatible with ISA `b` only if it's a subset of `b`
+        return a_isa ≤ b_isa
+    end
+end
+
+function augment_microarchitecture!(platform::Platform)
+    haskey(platform, "march") && return platform
```

Comment:
You might still need to call `set_compare_strategy` since I believe we can get here by just parting the triple

Reply:
To be clear, you're suggesting to apply `set_compare_strategy!(platform, "march", march_comparison_strategy)` before returning `platform`?

Reply:
Yes.